File: Unit4G1.ts
/**
 * @author William Provost
 * @version 1.0.0
 * @date 2025-11-30
 * @fileoverview This program finds average temperature, using arrays.
 */

// constants and variables
const temperature1: number[] = new Array(5); // Creates an array of 5 elements
let sum1: number = 0;

// get five temperatures from user
for (let counter = 0; counter < 5; counter++) {
  const userInput = prompt(`Enter temperature ${counter + 1}: `) || "0";
  const oneTemperature1 = parseFloat(userInput);
  temperature1[counter] = oneTemperature1;
}

// calculate sum using a for loop
for (let counter = 0; counter < temperature1.length; counter++) {
  sum1 += temperature1[counter];
}

// calculate average to two decimal places
// toFixed(2) returns a string, so we keep the const type as const
const average1 = (sum1 / temperature1.length).toFixed(2);

// display all temperatures using a for loop
console.log("\nThe temperatures are:");
for (let counter = 0; counter < temperature1.length; counter++) {
  console.log(`Temperature ${counter + 1}: ${temperature1[counter]}°C`);
}

// display the average
console.log(`\nThe average temperature is: ${average1}°C`);

console.log("\nDone.");

File: Unit4G1.go
// Author: Mr Coxall
// Version: 1.0.0
// Date: 2025-01-01
// Fileoverview: This program finds average temperature, using arrays.

package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	// variables
	// Array of 5 floating-point numbers
	var temperature [5]float64
	var scanner = bufio.NewScanner(os.Stdin)
	var sum float64 = 0

	// get five temperatures from user
	for counter := 0; counter < 5; counter++ {
		fmt.Printf("Enter temperature %d: ", counter+1)
		scanner.Scan()
		userInput := scanner.Text()
		
		// Parse the user input string into a float64
		oneTemperature, _ := strconv.ParseFloat(strings.TrimSpace(userInput), 64)
		temperature[counter] = oneTemperature
	}

	// calculate sum using a for loop
	for counter := 0; counter < len(temperature); counter++ {
		sum += temperature[counter]
	}

	// calculate average to two decimal places
	// Need to cast len(temperature) to float64 for division
	average := sum / float64(len(temperature))

	// display all temperatures using a for loop
	fmt.Println("\nThe temperatures are:")
	for counter := 0; counter < len(temperature); counter++ {
		// %d is integer, so we use %f for float
		fmt.Printf("Temperature %d: %.0f°C\n", counter+1, temperature[counter]) 
	}

	// display the average
	// %.2f formats the float to two decimal places
	fmt.Printf("\nThe average temperature is: %.2f°C\n", average)

	fmt.Println("\nDone.")
}
